### ISSUE TITLE
Use CLAP cosine similarity for mood scores (option 4 follow-up to #88)

### ISSUE THREAD

**Issue by Sxx7** [Sxx7/GrooveIQ#98]
## Background

The composite valence fix in #88 is a heuristic stack of weights over three EffNet binary heads. It works (mean 0.40, real spread, ranker can use it), but it's a calibration dance: each binary head has its own bias, and the weights only happen to be sensible for the current library.

While looking at how [AudioMuse-AI](https://github.com/NeptuneHub/AudioMuse-AI) handles the same problem, they ditched the per-label binary classifier approach entirely for the "feel" dimensions and use **CLAP text-audio cosine similarity** instead. Their `OTHER_FEATURE_LABELS` are literally identical to ours: `['danceable', 'aggressive', 'happy', 'party', 'relaxed', 'sad']`. The implementation is ~20 lines:

```python
# AudioMuse-AI tasks/clap_analyzer.py:788-810
def compute_other_features_from_clap(audio_embedding, label_embeddings):
    result = {}
    for label, text_emb in label_embeddings.items():
        # Cosine similarity = dot product for L2-normalized vectors
        similarity = float(np.dot(audio_embedding, text_emb))
        # Map [-1, 1] → [0, 1]
        result[label] = (similarity + 1.0) / 2.0
    return result
```

This is the option-4 \"different model\" path I dismissed in the #88 issue as too expensive. It's not — we already have everything we need.

## Why this is cheap for us

Already in place:
- `TrackFeatures.clap_embedding` column ([app/models/db.py:222](app/models/db.py:222)) stores 512-dim L2-normalised CLAP audio vectors
- `app.services.clap_text.encode_text()` produces 512-dim L2-normalised CLAP text vectors with caching
- `app.services.clap_backfill.backfill_clap_embeddings()` exists for backfilling tracks that don't have one yet

What's missing:
1. A small helper analogous to AudioMuse-AI's `compute_other_features_from_clap` — pre-encode our 5 mood labels once at startup, dot-product against `clap_embedding` per track
2. A migration path: where `clap_embedding IS NOT NULL`, prefer CLAP-derived mood scores; fall back to the EffNet composite from #88 elsewhere

## CLAP coverage caveat (important)

On local test server today: **12,551 / 93,739 tracks have a CLAP embedding (13.4%)**. So this can't fully replace the EffNet path until the CLAP backfill catches up. Two ways to handle that:

- **A. Hybrid**: write CLAP-derived `valence` / mood scores when `clap_embedding IS NOT NULL`, fall back to the EffNet composite otherwise. Mixed-source values across the library, but always populated.
- **B. Gate on CLAP coverage**: only switch over after CLAP backfill is ≥95% complete, then bump `ANALYSIS_VERSION` once and reprocess. Cleaner data, but requires a longer wait and a backfill push.

A. is probably the right call given how long full CLAP analysis takes — the EffNet composite is already producing usable values, so a slow rollout is fine.

## What this gets us

CLAP cosine similarity over a 512-dim shared text/audio embedding space gives smooth, well-spread scores by construction — no per-label binary head to compress at one extreme. Same labels we display today (`happy`, `sad`, `aggressive`, `relaxed`, `party`), but each is a continuous \"how close is this audio to the concept of X\" rather than a softmax probability over a tiny binary classifier.

It also opens a future door: once mood scores are dot-products against CLAP text prompts, adding a new mood label (e.g. `nostalgic`, `dark`, `uplifting`) is one line — no new model, no re-analysis. And `valence` becomes a real composition over CLAP scores rather than a hand-tuned weight stack.

## Out of scope here

- CLAP coverage backfill itself — that's an existing pipeline ([app/services/clap_backfill.py:32](app/services/clap_backfill.py:32)), this issue just consumes it
- The EffNet composite from #88 stays as the fallback / non-CLAP path
- AudioMuse-AI's MusiCNN multi-label \"social tag\" approach (their primary `mood_vector` of top-N Last.fm tags like `rock`, `80s`, `male vocalists`) — different concept, not what we want here

## Useful queries

```sql
-- CLAP coverage by analysis_version
SELECT analysis_version,
       COUNT(*) AS total,
       SUM(CASE WHEN clap_embedding IS NOT NULL THEN 1 ELSE 0 END) AS with_clap
FROM track_features
GROUP BY analysis_version;

-- After implementation: compare old vs new valence on the CLAP-covered subset
-- (would need to keep effnet-composite valence side-by-side temporarily)
```

**Comment by Sxx7** [comment]
Closing — implemented (b8ab439), verified on prod (local test server), then reverted (ac7da5c) after the quality numbers came back worse than #88.

## What we tried

Built `app/services/clap_mood.py` exactly as the issue described: 5 mood-label text prompts encoded once via `clap_text.encode_text`, dot-product against the stored 512-dim CLAP audio embedding, `(sim+1)/2` to map to [0,1], same `mood_tags` / `valence` shape the analysis worker writes today. Wired it into `pool.analyze()` post-processing (kept the text encoder out of every worker subprocess) and into the CLAP backfill so coverage would grow track-by-track. 22 unit tests, all green; ruff clean; deployed cleanly to prod.

## Why it's not shipping

Sampled 500 random CLAP-covered tracks on prod under four normalisation schemes:

| Scheme | valence mean | std | range | spread |
|---|---|---|---|---|
| **`(sim+1)/2`** (issue's proposal) | 0.557 | **0.017** | [0.483, 0.596] | **0.113** |
| #88 EffNet composite (status quo) | ~0.40 | ~0.10 | ~[0, 0.85] | ~0.40 |
| Per-label z-score → sigmoid | 0.501 | 0.092 | [0.286, 0.724] | 0.438 |
| Per-label percentile rank | 0.500 | 0.130 | [0.255, 0.818] | 0.563 |

The vanilla cosine approach produces *worse* spread than what #88 already gives us. Music-shaped audio always has positive cosine with music-shaped text concepts, so all 5 mood prompts cluster in [0.55, 0.70] per track (mean per-track top-vs-bottom mood gap = 0.014). Tried four prompt strategies — none meaningfully changes this.

## AudioMuse-AI re-examined

The issue cited AudioMuse-AI's `compute_other_features_from_clap` (`tasks/clap_analyzer.py:788-810`) as inspiration. After reading their code rather than the snippet:

1. **Their math is byte-for-byte identical to ours** — same `(sim+1)/2`, no calibration, no temperature, no softmax across labels.
2. **They have the exact same compression problem.** Their issue [#480](https://github.com/NeptuneHub/AudioMuse-AI/issues/480) has a verbatim row from a real library: `aggressive:0.61, happy:0.57, party:0.57, relaxed:0.55, sad:0.56` — per-track range 0.06, even tighter than ours.
3. **They never recognised it as a calibration problem.** Their fix was a hardcoded `>= 0.6` threshold + `ORDER BY <mood> DESC` in their MCP query layer — they work around the compression by relying on relative ordering that survives.
4. **They weight CLAP mood scores at literal zero** in their main downstream consumer (`SCORE_WEIGHT_OTHER_FEATURE_DIVERSITY = 0.0`, `SCORE_WEIGHT_OTHER_FEATURE_PURITY = 0.0` in their evolutionary clustering).
5. **They keep a parallel "real" mood pipeline** (50-class MusiCNN `mood_vector` at full sigmoid range) that does the actual mood semantics. CLAP `other_features` is reserved for boolean keyword filters and a dashboard pie chart.
6. **They store rounded to 2 decimals** (`f"{k}:{d.get(k, 0.0):.2f}"`) — actively destructive given the small spread CLAP does produce.
7. **The `mood_centroids_real_080_clap.json` file is mislabelled** — the centroids inside are 200-dim MusiCNN vectors, not CLAP. Used only as anchor seeds in their "Song Alchemy" recommender, never for mood scoring.

The "smooth, well-spread scores by construction" claim in this issue was a misread of their code. Their architecture survives the compression because they have *another* mood pipeline doing the real work. We don't.

## Realistic paths forward (if anyone wants to revisit)

- **Calibration**: per-label z-score across the library + sigmoid recovers ~0.44 spread (comparable to #88). Costs a periodic refit and a persisted `(mean, std)` artifact per label. ~50 lines of additional code.
- **Threshold-only consumer**: keep CLAP mood for relative ordering and `>= 0.6`-style boolean filters (à la AudioMuse-AI), leave `valence` on the EffNet composite. No spread improvement, but adds a useful relative-ordering signal.
- **Multi-prompt averaging per label** (e.g. 5 prompts per mood, average their cosines) — sometimes helps with prompt brittleness, doesn't fix the music-baseline compression.

## Net result

#88's composite stays as the production valence path. It's imperfect (party-dominated, ad-hoc weights) but it produces real spread that the LightGBM ranker can use.
